.github/workflows/cool.yml

```diff
@@ -1,182 +1,213 @@
-      - name: Push Docker Image
-        run: docker push $ECR_URI/$IMAGE_REPO:${{steps.vars.outputs.tag}}
-        if: github.event.client_payload.trigger == env.IMAGE_REPO || github.event.action == 'cool_production'
+        run: aws ecr get-login-password --region $REGION | docker login --username AWS --password-stdin $ECR_URI
 
-  terraform:
-    runs-on: ubuntu-latest
-    needs: [build_ui, build_api, setenv]
-    env:
-      ENVIRONMENT: ${{ needs.setenv.outputs.ENVIRONMENT }}
-    steps:
+      # Push UI
+      - name: Push UI
+        run: docker push $ECR_URI/$UI_IMAGE_REPO:$UI_TAG
+        if: env.BUILD_UI == 'true'
 
-      # Checkout domain-manager repo for running terraform
-      - uses: actions/checkout@v2
-        with:
-          path: domain-manager
+      # Push API
+      - name: Push API
+        run: docker push $ECR_URI/$API_IMAGE_REPO:$API_TAG
+        if: env.BUILD_API == 'true'
 
-      # Install Terraform
-      - uses: hashicorp/setup-terraform@v1
+      # Set default AWS credentials
+      - uses: aws-actions/configure-aws-credentials@v1
         with:
-          terraform_version: 0.12.30
-      - name: Verify Terraform version
-        run: terraform --version
+          aws-access-key-id: ${{ secrets.COOL_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.COOL_AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ env.REGION }}
 
       # Initialize Terraform
       - name: Terraform init
-        working-directory: ./domain-manager/cool
+        working-directory: ./cicd/cool
         run: terraform init -input=false
-
+      
       # Select terraform workspace
       - name: Terraform Workspace
-        working-directory: ./domain-manager/cool
+        working-directory: ./cicd/cool
         run: |
           terraform workspace select $ENVIRONMENT
           terraform workspace show
 
       # Validate Terraform
       - name: Terraform validation
-        working-directory: ./domain-manager/cool
+        working-directory: ./cicd/cool
         run: terraform validate
 
       # Sometimes this just needs run again...
       - name: Terraform init 2
-        working-directory: ./domain-manager/cool
+        working-directory: ./cicd/cool
         run: terraform init -input=false
 
-      # Apply terraform
+      # Plan terraform
       - name: Terraform Plan
-        working-directory: ./domain-manager/cool
-        run: terraform plan -input=false -var-file ./vars/$ENVIRONMENT.tfvars -var="api_image_tag=${{needs.build_api.outputs.tag}}" -var="ui_image_tag=${{needs.build_ui.outputs.tag}}"
-
+        working-directory: ./cicd/cool
+        run: |
+          terraform plan \
+            -input=false \
+            -var-file ./vars/$ENVIRONMENT.tfvars \
+            -var="api_image_tag=$API_TAG" \
+            -var="ui_image_tag=$UI_TAG" \
+            -var="account_id=$ACCOUNT_ID" \
+            -var="shared_services_account_id=$SHARED_SERVICES_ACCOUNT_ID"
```

Comment:
What is the reason for running `terraform plan` here?

Reply:
@dav3r The purpose of running terraform plan there is mostly for the pull requests. It is also helpful for when it runs `terraform apply` automatically in the pipeline as `terraform plan` gives more details as far as specifics of what changes. Look at the [latest pipeline run as an example](https://github.com/cisagov/domain-manager-cicd/runs/2632191008?check_suite_focus=true).

Reply:
Got it, thanks.  I never use the `-auto-approve` switch, which apparently does not display the plan before executing it.